patches/tModLoader/Terraria/ModLoader/LocalizationLoader.cs

```diff
@@ -323,7 +327,7 @@ private static void UpdateLocalizationFilesForMod(Mod mod, string outputPath = n
 		};
 
 		// TODO: Maybe optimize to only recently built?
-		string sourceFolder = outputPath ?? Path.Combine(ModCompile.ModSourcePath, mod.Name);
+		string sourceFolder = outputPath ?? BuildProperties.ReadModFile(mod.File).modSource;
```

Comment:
Bleh, really second guessing keeping the `BuildProperties` on `LocalMod` rather than `Mod`, but anyway. Please copy `modSource` to a new property in `Mod`, call it `SourceFolder` perhaps, and assign it in `AssemblyManager.Instantiate`

Reply:
Maybe copy it only `if (Directory.Exists(modSource))` just so people who are looking through `Mod` instances in their debugger don't get nosy